modules/local/transform/main.nf

```diff
@@ -0,0 +1,29 @@
+process TRANSFORM_METADATA {
+    tag "transform metadata"
+    label 'process_single'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
```

Comment:
This is just a note and not something you have to do, but I wonder if there's an easier way to find smaller containers with just the packages you want in them (pandas, numpy). I know there's info on the Galaxy admins page on the process they use: https://docs.galaxyproject.org/en/latest/admin/special_topics/mulled_containers.html

Using the staramr container here is okay though, even if we're not using all of staramr. I'm wondering if you could add a comment here describing why we're using the staramr container though?

Reply:
Mulled container repo is here: https://github.com/BioContainers/multi-package-containers and you can see what combinations they have in the [hashes.tsv](https://github.com/BioContainers/multi-package-containers/blob/master/combinations/hash.tsv) file. I've used it a few times

Reply:
Ideally we want Python>=3, pandas>=2, numpy>=2. I found these:

```
python=3.11.4,pandas=2.0.3,numpy=1.25.1,scikit-learn=1.3.0,click=8.1.6,matplotlib=3.7.2
numpy=1.25.2,pandas=2.0.3,bx-python=0.10.0,gzip=1.12
pandas=2.2.0,numpy=1.26.3,statsmodels=0.14.1
```

I might be able to get away with an earlier version of numpy? StarAMR is running pandas=2.2.3 and numpy=2.1.3.

Reply:
Thanks so much @DarianHole  and @emarinier.

Let's stick to StarAMR container right now. When we've finished up all the additional stories/functionality for this pipeline we can maybe revisit the best container for this.